src/screens/Server.js

```diff
@@ -87,26 +125,63 @@ class Server extends React.Component {
       return;
     }
 
-    this.setState({ loading: true, errorMessage: '', selectedServer: newServer });
-    const currentServer = hathorLib.helpers.getServerURL();
-    // Update new server in local storage
-    hathorLib.wallet.changeServer(newServer);
-    hathorLib.versionApi.getVersion((data) => {
-      // If the user selected a testnet server, we show a modal to confirm the operation
-      if (data.network !== 'mainnet') {
+    this.setState({
+      loading: true,
+      errorMessage: '',
+      selectedServer: newBaseServer,
+      selectedWsServer: newWsServer,
+    });
+
+    const currentServer = this.props.useWalletService ?
+      hathorLib.config.getWalletServiceBaseUrl() : 
+      hathorLib.config.getServerUrl();
+
+    const currentWsServer = this.props.useWalletService ?
+      hathorLib.config.getWalletServiceBaseWsUrl() :
+      '';
+
+    // Update new server in storage and in the config singleton
+    this.props.wallet.changeServer(newBaseServer);
+
+    // We only have a different websocket server on the wallet-service facade, so update the config singleton
+    if (this.props.useWalletService) {
+      this.props.wallet.changeWsServer(newWsServer);
+    }
+
+    try {
+      const versionData = await this.props.wallet.getVersionData();
+
+      if (versionData.network !== 'mainnet') {
+        const network = versionData.network;
+        let selectedNetwork = network;
+
+        // Network might be 'testnet-golf' or 'testnet-charlie'
+        if (network.startsWith('testnet')) {
+          selectedNetwork = 'testnet';
+        }
+
+        this.setState({
+          selectedNetwork,
+        });
+
         // Go back to the previous server
         // If the user decides to continue with this change, we will update again
-        hathorLib.wallet.changeServer(currentServer);
+        this.props.wallet.changeServer(currentServer);
+        if (this.props.useWalletService) {
+          hathorLib.config.setWalletServiceBaseWsUrl(currentWsServer);
+        }
         $('#alertModal').modal('show');
-        this.setState({ loading: false });
```

Comment:
Why did you remove this? The loading will continue showing with the modal?

Reply:
Not sure why I removed it, makes sense to stop the loading when we show the modal

Fixed in ce9279adb876c243c4294a16ae7c4617c05dbe03